rpc/cloudagent/network/loadbalancer/moc_cloudagent_loadbalancer.proto

```diff
@@ -27,10 +27,39 @@ message LoadbalancerInboundNatRule {
 	Protocol protocol = 4;
 }
 
+message LoadbalancerOutboundNatRule {
+	string name = 1;
+	uint32 frontendPort = 2;
+	uint32 backendPort = 3;
+	Protocol protocol = 4;
+}
+
 message LoadBalancingRule {
 	uint32 frontendPort = 1;
 	uint32 backendPort = 2;
 	Protocol protocol = 3;
+	ProbeReference probeRef = 4; // Update based on chat with Nithin
+}
+
+message Probe {
+	uint32 intervalInSeconds = 1;
+	uint32 numberOfProbes = 2;
+	Protocol protocol = 3;
+	uint32 port = 4; 
+	repeated LoadBalancingRule loadbalancingrules = 5; //read-only
```

Comment:
>LoadBalancingRule [](http://example.com/codeflow?start=10&length=17)

readonly ref like this is a tricky one. MOC currently doesnt have a concept of readonly ref. are you planning to populate this or is this for future?